### PR TITLE
stacks and queues

### PR DESCRIPTION
**Pull request by paronicholas**
- [x] Top-level README “Table of Contents” is updated
- [x] Feature tasks for this challenge are completed
- [x] Unit tests written and passing
    - [x] “Happy Path” - Expected outcome
    - [x] Expected failure
    - [x] Edge Case (if applicable/obvious)
- [x] README for this challenge is complete
    - [x] Summary, Description, Approach & Efficiency, Solution, Resources
    - [x] Link to code
    - [x] Link to tests

### CODE REVIEW

File: code401challenges/src/main/java/stacksAndQueues/Node.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  public void setNext(Node<T> node) {
+    this.next = node;
+  }
+
+  public Node<T> getNext() {
+    return this.next;
+  }
+
+  public String toString() {
+    return (String)this.value;
```
Comment (evanslaton) [inline code401challenges/src/main/java/stacksAndQueues/Node.java]: I don't know if this will work if the value isn't a String. Maybe `String.valueOf()` would work?

File: code401challenges/src/main/java/stacksAndQueues/Queue.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  public void enqueue(T value) {
+    Node<T> newNode = new Node<>();
+    newNode.value = value;
+
+    if (front == null) {
+      this.front = newNode;
+      this.back = newNode;
+      return;
```
Comment (evanslaton) [inline code401challenges/src/main/java/stacksAndQueues/Queue.java]: Instead of having `return` sprinkled throughout your code, why not use an `else`?

File: code401challenges/src/main/java/stacksAndQueues/Queue.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    this.back.next = newNode;
+    this.back = newNode;
+  }
+
+  public T dequeue() {
+    if (this.front == null) {
```
Comment (evanslaton) [inline code401challenges/src/main/java/stacksAndQueues/Queue.java]: Since this conditional also exists in duplicate in the `peek` method. Could you just call `peek` here instead?

File: code401challenges/src/main/java/stacksAndQueues/Stack.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    top = newNode;
+  }
+
+  public T pop() {
+    if (top.value == null) {
+      throw new NullPointerException();
+    }
+    Node<T> poppedNode = this.top;
+    this.top = poppedNode.next;
+    poppedNode.next = null;
```
Comment (evanslaton) [inline code401challenges/src/main/java/stacksAndQueues/Stack.java]: This line is extra. Since the reference to poppedNode vanishes once the method finishes the node will be garbage collected
